Lib/fontTools/ttLib/tables/DefaultTable.py

```diff
@@ -40,7 +42,7 @@ def __repr__(self):
         return "<'%s' table at %x>" % (self.tableTag, id(self))
 
     def __eq__(self, other):
-        if type(self) != type(other):
+        if not isinstance(self, type(other)):
```

Comment:
This is unrelated to type hints and changes the meaning of the code. The current code is written as intended: only compare when the types match exactly.

Reply:
You are right. I tried to avoid changes like this. I’ll remove it.